core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/generation/OicrLibraryAliasGenerator.java

```diff
@@ -0,0 +1,51 @@
+package uk.ac.bbsrc.tgac.miso.core.service.naming.generation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Library;
+import uk.ac.bbsrc.tgac.miso.core.exception.MisoNamingException;
+
+// TODO: fix/finish - this is the alias generation code as it was (not working) before naming scheme refactor (GLT-551)
+public class OicrLibraryAliasGenerator implements NameGenerator<Library> {
+
+  private static final String sampleRegex = "^([A-Z\\d]{3,5}_\\d{3,5}_(?:[A-Z][a-z]|nn)_[A-Zn])_.*";
+  private static final Pattern samplePattern = Pattern.compile(sampleRegex);
+
+  @Override
+  public String generate(Library library) throws MisoNamingException {
+    if (library.getSample() != null) {
+      Matcher m = samplePattern.matcher(library.getSample().getAlias());
+
+      if (m.matches()) {
+        // try {
+        // int numLibs = requestManager.listAllLibrariesBySampleId(l.getSample().getId()).size();
```

Comment:
This is the library alias generation code more-or-less copy/pasted from what existed before. I did not fix the generator, which didn't work before. I'm not sure if the code that's commented out is useful. I will make sure there is a ticket open for getting Library alias generation working though